src/models/parsers/buildin/discord.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    CategoryChannel,
+    Channel,
+    DMChannel,
+    MessageMentions,
+    NewsChannel,
+    Role,
+    StoreChannel,
+    TextChannel,
+    User,
+    VoiceChannel,
+} from "discord.js";
+import { distinct } from "../../../utils/array";
+import {
+    SnowflakeParseError,
+    UnresolvedChannelParseError,
+    UnresolvedRoleParseError,
+    UnresolvedUserParseError,
+    WrongTypeChannelParseError,
+} from "../errors/discord";
+import { Parser } from "../Parser";
+import { ParsingContext } from "../ParsingContext";
+
+/** @internal */
+const ID_PATTERN = /^\d{17,21}$/;
+
+export class UserParser extends Parser<User> {
+    constructor() {
+        super("$user$", 1);
+    }
+
+    protected parse(context: ParsingContext): User {
+        const str = context.nextString();
+        const idResult = MessageMentions.USERS_PATTERN.exec(str);
+        MessageMentions.USERS_PATTERN.lastIndex = 0;
```

Comment:
We should use these parttern instead:
- not global (no need `USER_PATTERN.lastIndex = 0`)
- single value pattern
```ts
const USERS_PATTERN = /^<@!?(\d{17,19})>$/; // <@81440962496172032>
const ROLES_PATTERN = /^<@&(\d{17,19})>$/; // <@&297577916114403338>
const CHANNELS_PATTERN = /^<#(\d{17,19})>$/; // <#222079895583457280>
```